### PR TITLE
fix port allocate

### PR DESCRIPTION
**Pull request by zwwhdls**
### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/fluid-cloudnative/fluid/issues/2763

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/juicefs/transform.go [hunk 1]
```diff
@@ -195,16 +196,20 @@ func (j *JuiceFSEngine) allocatePorts(dataset *datav1alpha1.Dataset, runtime *da
...
+		j.Log.Error(err, "can't get available ports", "expected port num", exportWorkerPodNum+exportFusePodNum)
 		return err
 	}
 
 	index := 0
-	value.Worker.MetricsPort = &allocatedPorts[index]
+	if exportWorkerPodNum > 0 {
+		value.Worker.MetricsPort = &allocatedPorts[index]
+	}
 	index++
```
Comment (TrafalgarZZZ) [inline pkg/ddc/juicefs/transform.go:209]: Should not `index++` if exportWorkerPodNum==0.

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]

File: pkg/ddc/juicefs/transform.go [hunk 2]
```diff
@@ -176,16 +176,17 @@ func (j *JuiceFSEngine) allocatePorts(dataset *datav1alpha1.Dataset, runtime *da
 	// if not use hostnetwork then use default port
 	// use hostnetwork to choose port from port allocator
 
-	expectedPortNum := 2
+	exportWorkerPodNum := 1
```
Comment (TrafalgarZZZ) [inline pkg/ddc/juicefs/transform.go:179]: Is `export` a typo here? Did you mean `expected`?

File: pkg/ddc/juicefs/transform.go [hunk 3]
```diff
@@ -176,16 +176,17 @@ func (j *JuiceFSEngine) allocatePorts(dataset *datav1alpha1.Dataset, runtime *da
...
+	exportFusePodNum := 1
 	if !datav1alpha1.IsHostNetwork(runtime.Spec.Worker.NetworkMode) {
 		value.Worker.MetricsPort = &workerMetricsPort
```
Comment (cheyang) [inline pkg/ddc/juicefs/transform.go:182]: I think only using hostNetwork should need allocate metrics port.

File: pkg/ddc/juicefs/transform.go (hunk 1 above)
Comment (zwwhdls) [reply pkg/ddc/juicefs/transform.go]: thks. done.

File: pkg/ddc/juicefs/transform.go (hunk 2 above)
Comment (zwwhdls) [reply pkg/ddc/juicefs/transform.go]: yeah.

File: pkg/ddc/juicefs/transform.go [hunk 4]
```diff
@@ -176,17 +176,17 @@ func (j *JuiceFSEngine) allocatePorts(dataset *datav1alpha1.Dataset, runtime *da
 	// if not use hostnetwork then use default port
 	// use hostnetwork to choose port from port allocator
 
-	exportWorkerPodNum := 1
-	exportFusePodNum := 1
+	expectWorkerPodNum := 1
+	expectFusePodNum := 1
```
Comment (cheyang) [inline pkg/ddc/juicefs/transform.go:180]: I suggest using `expectWorkerPodNum, expectFusePodNum := 1, 1`

**Review by cheyang — APPROVED** [review]

**Review by TrafalgarZZZ — APPROVED** [review]
/lgtm

**Pull request merged**